edkrepo/common/common_repo_functions.py

```diff
@@ -832,11 +833,19 @@ def create_local_branch(name, patchset, global_manifest_path, manifest_obj, repo
                 repo.remotes.origin.fetch(patchset.fetch_branch, progress=GitProgressHandler())
             except:
                 raise EdkrepoFetchBranchNotFoundException(FETCH_BRANCH_DOES_NOT_EXIST.format(patchset.fetch_branch))
+            parent_patchsets = list()
             try:
-                parent_patchset = manifest_obj.get_patchset(patchset.parent_sha, patchset.remote)
-                repo.git.checkout(parent_patchset[2], b=name)
-            except:
-                if patchset.parent_sha in repo.tags:
+                parent_patchsets.append(manifest_obj.get_patchset(patchset.parent_sha, patchset.remote))
+                while True:
+                    # Allow daisy-chaining PatchSet.  Continue until KeyError is caught.
+                    parent_patchsets.append(manifest_obj.get_patchset(parent_patchsets[len(parent_patchsets)-1].parent_sha, parent_patchsets[len(parent_patchsets)-1].remote))
+                    if parent_patchsets.count(parent_patchsets[len(parent_patchsets)-1]) > 1:
+                        # Do not continue to branch creation step if a circular dependency is detected between multiple PatchSet.
+                        raise ValueError(PATCHSET_CIRCULAR_DEPENDENCY_ERROR.format(parent_patchsets[len(parent_patchsets)-1]))
+            except KeyError:
+                if parent_patchsets != list():
+                    repo.git.checkout(parent_patchsets[len(parent_patchsets)-1][2], b=name)

```

Comment:
This needs a try/except block similar to line 848-851 to handle the case of a branch and a tag having the same name.

Reply:
Confirmed there is a failure when the base patchset parent sha is both a tag name and a branch name at the same time.

Added a tag to the Edk2 repo named "master".
Created a test PatchSet "edk2_test_C" that is double-daisy-chained to a tag named "master".

    <PatchSet remote="Edk2Repo" name="edk2_test_C" parentSha="edk2_test_B" fetchBranch="master">
    </PatchSet>
    <PatchSet remote="Edk2Repo" name="edk2_test_B" parentSha="edk2_test_A" fetchBranch="master">
    </PatchSet>
    <PatchSet remote="Edk2Repo" name="edk2_test_A" parentSha="master" fetchBranch="master">
    </PatchSet>

When checking a combo that uses PatchSet "edk2_test_C", failed with the error message:
"fatal: ambiguous object name: 'master'' "